tests/test_worker.py

```diff
@@ -189,6 +189,18 @@ class Settings:
     assert '  X.XXs ! testing:retry max retries 5 exceeded' in log
 
 
+async def test_retry_lots_without_keep_result(arq_redis: ArqRedis, worker):
+    async def retry(ctx):
+        raise Retry()
+
+    await arq_redis.enqueue_job('retry', _job_id='testing')
+    worker: Worker = worker(functions=[func(retry, name='retry')], keep_result=0)
+    try:
```

Comment:
```suggestion
    with pytest.raises(MultiExecError, match='ERR invalid expire time in setex')
```
and remove the `except` clause.

Reply:
@samuelcolvin `MultiExecError` should not be raised. This test is check that we retry all attempts without worker crash.

Reply:
okay, sorry I misread.

Then best to remove the try/except completely.

Reply:
How about if I will add `keep_result=0` into `test_retry_lots` test case worker? It should work because we don't check job result for this one.

Reply:
I think best to keep this separate test, just remove the try/except.